daml-assistant/src/DAML/Assistant/Install.hs

```diff
@@ -183,19 +161,35 @@ installExtracted env@InstallEnv{..} sourcePath =
                 , walkOnDirectoryPost = \path ->
                     when (path /= addTrailingPathSeparator (unwrapSdkPath sourcePath)) $
                         setSdkFileMode path
-                , walkOnDirectoryPre = \_ -> pure ()
+                , walkOnDirectoryPre = const (pure ())
                 }
 
+        setupDamlPath damlPath
+
         let targetPath = defaultSdkPath damlPath sourceVersion
+
+        whenM (doesDirectoryExist (unwrapSdkPath targetPath)) $ do
+            unlessForce env $ do
+                throwIO $ assistantErrorBecause
+                    ("SDK version " <> versionToText sourceVersion <> " already installed. Use --force to overwrite.")
```

Comment:
Should we first remove the directory if we are doing a force install? It seems unintuitive to keep files from a previous installation if they do not exist in the new installation.

Reply:
They are removed a bit further down.